src/components/Contacts/ContactDetails/ContactDetailsTab/Tags/ContactTags.test.tsx

```diff
@@ -0,0 +1,161 @@
+const contactId = 'abc';
+const contactTags = ['tag1', 'tag2', 'tag3'];
+
+describe('ContactTags', () => {
+  it('should render with tags', () => {
+    const { getByText } = render(
+      <SnackbarProvider>
+        <GqlMockedProvider<UpdateContactTagsMutation>>
+          <MuiThemeProvider theme={theme}>
+            <ContactTags
+              accountListId={accountListId}
+              contactId={contactId}
+              contactTags={contactTags}
+            />
+          </MuiThemeProvider>
+        </GqlMockedProvider>
+      </SnackbarProvider>,
+    );
+    expect(getByText('tag1')).toBeInTheDocument();
+    expect(getByText('tag2')).toBeInTheDocument();
+    expect(getByText('tag3')).toBeInTheDocument();
+  });
+
+  it('should add a tag', async () => {
+    const cache = new InMemoryCache();
+    jest.spyOn(cache, 'writeQuery');
+    const data: ContactDetailsTabQuery = {
+      contact: {
+        __typename: 'Contact',
+        id: contactId,
+        tagList: contactTags,
+        people: {
+          nodes: [],
+        },
+        name: 'Lname, Fname',
+      },
+    };
+    const query = {
+      query: ContactDetailsTabDocument,
+      variables: {
+        accountListId,
+        contactId,
+      },
+      data,
+    };
+    cache.writeQuery(query);
+    const { getByRole } = render(
+      <SnackbarProvider>
+        <GqlMockedProvider<UpdateContactTagsMutation>
+          mocks={{
+            UpdateContactTags: {
+              updateContact: {
+                contact: {
+                  __typename: 'Contact',
+                  id: contactId,
+                  tagList: [...contactTags, 'tag4'],
+                  people: {
+                    nodes: [],
+                  },
+                  name: 'Lname, Fname',
+                },
+              },
+            },
+          }}
+          cache={cache}
+        >
+          <MuiThemeProvider theme={theme}>
+            <ContactTags
+              accountListId={accountListId}
+              contactId={contactId}
+              contactTags={contactTags}
+            />
+          </MuiThemeProvider>
+        </GqlMockedProvider>
+      </SnackbarProvider>,
+    );
+    userEvent.type(getByRole('textbox', { name: 'Tag' }), 'tag4{enter}');
+    await waitFor(() =>
+      expect(getByRole('textbox', { name: 'Tag' })).toHaveValue(''),
+    );
+
+    // TODO: Figure out why readQuery value is always null
+    // await waitFor(() =>
+    //   expect(cache.writeQuery).toHaveBeenCalledWith({
+    //     query: ContactDetailsTabDocument,
+    //     variables: {
+    //       accountListId,
+    //       contactId,
+    //     },
+    //     data: {
+    //       contact: {
+    //         __typename: 'Contact',
+    //         id: contactId,
+    //         tagList: [...contactTags, 'tag4'],
+    //         people: {
+    //           nodes: [],
+    //         },
+    //         name: 'Lname, Fname',
+    //       },
+    //     },
+    //   }),
+    // );
```

Comment:
@OzzieOrca Not sure what's up with this test. I've set it up similar to other mutations test, but the value of ```readQuery``` is always null. Therefore, the ```writeQuery``` logic never fires. But while debugging the test, I can see data in the cache
<img width="611" alt="Screen Shot 2021-05-27 at 3 06 53 PM" src="https://user-images.githubusercontent.com/39680460/119887873-b0ae8800-bf02-11eb-8d72-291f8c768b97.png">